### PR TITLE
Port over intervals

### PR DESCRIPTION
**Pull request by jeromekelleher**
Updated version of #2636 with care taken to have full information about code provenance.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
I've basically just done some cherry picking from your commits @hyanwong, so it should be the same as #2636 (including your email in the Git log for the commits)

File: python/tskit/tables.py [hunk 1]
```diff
@@ -657,41 +656,10 @@ def __str__(self):
         return util.unicode_table(rows, header=headers, row_separator=False)
 
     def _repr_html_(self):
-        """
```
Comment (hyanwong) [inline python/tskit/tables.py:660]: Maybe we don't want to delete this docstring, as it could be used in the documentation for the table class?

**Review by hyanwong — APPROVED** [review]
Thanks. That looks great. One minor comment

File: python/tskit/tables.py (hunk 1 above)
Comment (hyanwong) [reply python/tskit/tables.py:660]: Yes, I guess so. I can alway reinstate. It's not used in the docs ATM anyway (although the `_repr_html_` methods on tree sequences and trees *are* listed)

**Comment by hyanwong** [comment]
Do we want the documentation to appear in tskit now, rather than msprime? I assume so. I can sort that later if you like @jeromekelleher ?

**Comment by hyanwong** [comment]
#2679

File: python/tskit/util.py [hunk 2]
```diff
@@ -423,16 +432,55 @@ def unicode_table(rows, title=None, header=None, row_separator=True):
...
             )
             last_skipped = False
 
     out.append(f"╚{'╧'.join('═' * w for w in widths)}╝\n")
     return "".join(out)
 
 
+def html_table(rows, *, header):
+    """
+    Called e.g. by jupyter notebooks to render tables
```
Comment (benjeffery) [inline python/tskit/util.py:451]: This docstring isn't correct, as this method isn't directly called by jupyter.

**Review by benjeffery — APPROVED** [review]
LGTM, one nit about the docstring. Thanks for factoring out `truncate_rows` I should have done that with the original work!

File: python/tskit/util.py (hunk 2 above)
Comment (hyanwong) [reply python/tskit/util.py:451]: Yes, I think I wrongly moved that from the `_repr_html_` method (comment above). I guess it should be moved back. I don't think I have write access to this PR any more, as Jerome opened a new branch, so maybe you could do it @benjeffery ?

File: python/tskit/tables.py (hunk 1 above)
Comment (jeromekelleher) [reply python/tskit/tables.py:660]: Fixed

File: python/tskit/util.py (hunk 2 above)
Comment (jeromekelleher) [reply python/tskit/util.py]: Fixed.

Comment (hyanwong) [reply python/tskit/util.py]: Thanks a lot @jeromekelleher

**Comment by jeromekelleher** [comment]
Huh, no idea what this docs build failure is about - have you seen this one before @benjeffery?

**Comment by benjeffery** [comment]
> Huh, no idea what this docs build failure is about - have you seen this one before @benjeffery?

Yep, first happened last night and is fixed in the Py3.11 PR that I'm merging imminently when codecov want to work.

**Comment by benjeffery** [comment]
Rebased and conflicts fixed.

**Comment by benjeffery** [comment]
@mergifyio rebase

**Comment by benjeffery** [comment]
@mergifyio refresh

**Pull request merged**